src/node_sets.cpp

```diff
@@ -176,6 +176,120 @@ class NodeSetBasicNodeIds: public NodeSetRule
     Selection::Values values_;
 };
 
+class NodeSetBasicOperatorString: public NodeSetRule
+{
+  public:
+    explicit NodeSetBasicOperatorString(const std::string attribute,
+                                        const std::string op,
+                                        const std::string value)
+        : op_(string2op(op))
+        , attribute_(attribute)
+        , value_(value) {}
+
+    Selection materialize(const detail::NodeSets& /* unused */,
+                          const NodePopulation& np) const final {
+        return np.regexMatch(attribute_, value_);
+    }
+
+    std::string toJSON() const final {
+        return fmt::format(R"("{}": {{ "{}": "{}" }})", attribute_, op2string(op_), value_);
+    }
+
+    enum class Op {
+        regex = 1,
+    };
+
+    static Op string2op(const std::string& s) {
+        if (s == "$regex") {
+            return Op::regex;
```

Comment:
So… in theory we could have a `$glob` with a small translation layer from glob→regular expression?

In the recipe we use glob matching, and it may be beneficial to offer a homogeneous interface to query stuff.